proxy/src/telemetry/metrics/latency.rs

```diff
@@ -0,0 +1,180 @@
+#![deny(missing_docs)]
+use std::{ops, slice, u32};
+use std::default::Default;
+use std::time::Duration;
+
+/// The number of buckets in a  latency histogram.
+pub const NUM_BUCKETS: usize = 26;
+
+static BUCKETS: [Latency; NUM_BUCKETS] = [
+    // The controller telemetry server creates 5 sets of 5 linear buckets
+    // each:
+    // TODO: it would be nice if we didn't have to hard-code each
+    //       individual bucket and could use Rust ranges or something.
+    //       However, because we're using a raw fixed size array rather
+    //       than a vector (as we don't ever expect to grow this array
+    //       and thus don't _need_ a vector) we can't concatenate it
+    //       from smaller arrays, making it difficult to construct
+    //       programmatically...
```

Comment:
If you want to construct this programmatically you can statically allocate the array and then programmatically fill it in:
```
for i in 1..6 {
  for j in 1..6 BUCKETS[5*i + j] = j*10.pow(i);
}
```
But I think listing the values out in the code is fine.

Reply:
Hmm, but that requires `BUCKETS` to be a static _mut_, which I'd rather avoid.